test/e2e/framework/nodes.go

```diff
@@ -0,0 +1,40 @@
+package framework
+
+import (
+	"strconv"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (f *Framework) FindGatewayAndNonGatewayNodes(cluster ClusterIndex) (gateway *v1.Node, nonGateway *v1.Node) {
```

Comment:
nit, nothing functional:

May be just GetNodes? (current function name seems overly verbose, and the fact that we will do

gateway, nonGateway := nodes.GetNodes() (even nodes.Get() ?)

would be verbose enough during usage?

Reply:
It's more than simply GetNodes - specialized to find 2 specific nodes so I prefer the verbosity for clarity.

Reply:
+1 to GetNodes

Reply:
OK - I'll shorten it but I think it should be `FindNodes`, as it's a search - `GetNodes` implies a lookup.